pkg/cmd/cli/cmd/create_route.go

```diff
@@ -0,0 +1,390 @@
+			cmdutil.CheckErr(err)
+		},
+	}
+
+	cmdutil.AddValidateFlags(cmd)
+	cmdutil.AddOutputFlagsForMutation(cmd)
+
+	return cmd
+}
+
+// CreatePassthroughRoute implements the behavior to run the create passthrough route command.
+func CreatePassthroughRoute(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []string) error {
+	oc, kc, err := f.Clients()
+	if err != nil {
+		return err
+	}
+
+	ns, _, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+	route, err := unsecuredRoute(cmd, kc, ns, args)
+	if err != nil {
+		return err
+	}
+
+	route.Spec.Host = cmdutil.GetFlagString(cmd, "hostname")
+
+	route.Spec.TLS = new(api.TLSConfig)
+	route.Spec.TLS.Termination = api.TLSTerminationPassthrough
+
+	route, err = oc.Routes(ns).Create(route)
+	if err != nil {
+		return err
+	}
+	mapper, typer := f.Object()
+	resourceMapper := &resource.Mapper{
+		ObjectTyper:  typer,
+		RESTMapper:   mapper,
+		ClientMapper: resource.ClientMapperFunc(f.Factory.RESTClient),
+	}
+	info, err := resourceMapper.InfoForObject(route)
+	if err != nil {
+		return err
+	}
+	shortOutput := cmdutil.GetFlagString(cmd, "output") == "name"
+	cmdutil.PrintSuccess(mapper, shortOutput, out, info.Mapping.Resource, info.Name, "created")
+	return nil
+}
+
+const (
+	reencryptRouteLong = `
+Create a route that uses reencrypt TLS termination
+
+Specify the service that the new route should expose via the --expose flag.
+A destination CA certificate is needed for reencrypt routes, specify one 
+with the --dest-ca-cert flag.`
+
+	reencryptRouteExample = `  # Create a route named "my-route" that exposes the frontend service.
+  $ %[1]s create route reencrypt my-route --expose frontend --dest-ca-cert pod.cert
+
+  # Create a reencrypt route that exposes the frontend service and re-use
+  # the service name as the route name.
+  $ %[1]s create route reencrypt --expose frontend --dest-ca-cert pod.cert`
+)
+
+// NewCmdCreateReencryptRoute is a macro command to create a reencrypt route.
+func NewCmdCreateReencryptRoute(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "reencrypt [NAME] --dest-ca-cert=FILENAME --expose=SERVICE",
+		Short:   "Create a route that uses reencrypt TLS termination",
+		Long:    reencryptRouteLong,
+		Example: fmt.Sprintf(reencryptRouteExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			err := CreateReencryptRoute(f, out, cmd, args)
+			cmdutil.CheckErr(err)
+		},
+	}
+
+	cmdutil.AddValidateFlags(cmd)
+	cmdutil.AddOutputFlagsForMutation(cmd)
+	cmd.Flags().String("path", "", "Path that the router watches to route traffic to the service.")
+	cmd.Flags().String("cert-file", "", "Provide certificate contents.")
+	cmd.Flags().String("key-file", "", "Provide key contents.")
+	cmd.Flags().String("ca-cert-file", "", "Provide cert authority certificate contents.")
+	cmd.Flags().String("dest-ca-cert", "", "Provide the contents of the ca certificate of the final destination.")
```

Comment:
Wrong link

this is it 
https://github.com/openshift/origin/blob/a1c70def8b38b98a5142e58fece9e25b840055c1/pkg/route/api/validation/validation.go#L95